### PR TITLE
Issue #9524: Update MissingJavadocMethodCheckTest to use unique input files

### LINKED ISSUE #9524: update MissingJavadocMethodCheckTest to use unique Input file in each test method
MissingJavadocMethodCheckTest.java file reuse some Input files few time in few different test method.
Expected: each test method in Test class use it own Input file and do not share it with any other test method.

more details - https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-create-Input-files-for-tests

### PR DESCRIPTION
**Pull request by bossever**
Fixes #9524 

1. Deleted two duplicate unit tests, changes done in seperate commit.
2. Removed Javadoc @author tag in `InputMissingJavadocMethodScopeAnonInner` and `InputMissingJavadocMethodExtendAnnotation` - see https://github.com/checkstyle/checkstyle/issues/2506#issue-114532272

### CODE REVIEW

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodInterfacePrivateMethod.java [hunk 1]
```diff
@@ -3,6 +3,9 @@
 
 package com.puppycrawl.tools.checkstyle.checks.javadoc.missingjavadocmethod;
 
+/* Config:
+ * scope = public
```
Comment (nmancus1) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodInterfacePrivateMethod.java:7]: I do not know if we have reached a consensus on this yet, but I always enclose check  properties in `"`, i.e. `scope = "public"`.  One day in the future we will derive configs for UT's  from these comments, and some delimiter will help with that. (This goes for all below configs, too.)

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc2.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
+package com.puppycrawl.tools.checkstyle.checks.javadoc.missingjavadocmethod;
+
+/* Config:
+ * scope = Scope.PROTECTED.getName()
```
Comment (nmancus1) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc2.java:4]: Why are some comments/ check properties shown as methods invocations? I would stick with `scope = "protected"`.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc3.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
+package com.puppycrawl.tools.checkstyle.checks.javadoc.missingjavadocmethod;
+
+/* Config:
+ * scope = Scope.PRIVATE.getName()
+ * excludeScope = Scope.PROTECTED.getName()
```
Comment (nmancus1) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc3.java:5]: Same as above.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
+////////////////////////////////////////////////////////////////////////////////
+// Test case file for checkstyle.
+// Created: 2021
+////////////////////////////////////////////////////////////////////////////////
```
Comment (nmancus1) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java:4]: We do not use this header for tests anymore, you can remove.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+package com.puppycrawl.tools.checkstyle.checks.javadoc.missingjavadocmethod;
+
+/* Config:
+ * scope = Scope.NOTHING.getName()
+ */
+public class InputMissingJavadocMethodPublicOnly2 // ignore - need javadoc
+{
+    private interface InnerInterface // ignore - when not relaxed about Javadoc
```
Comment (nmancus1) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java:12]: Should this be `ignore - when relaxed about javadoc`?  I think if we  were NOT relaxed, we would show violations for these missing javadocs.  Does that make sense?

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly3.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
+////////////////////////////////////////////////////////////////////////////////
+// Test case file for checkstyle.
+// Created: 2021
+////////////////////////////////////////////////////////////////////////////////
```
Comment (nmancus1) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly3.java:4]: Same as above.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    public void method(StringBuffer aA) // ok
+    {
+    }
+
+
+    /**
+     A param tag should not be required here when relaxed about Javadoc.
+     Writing a little documentation should not be worse than not
+     writing any documentation at all.
+     */
```
Comment (nmancus1) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java:96]: Just curious, why use the non-standard javadoc format here?

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner.java [hunk 8]
```diff
@@ -8,52 +8,54 @@
...
+     */
     private JButton mButton = new JButton();
 
     /**
-       anon inner in member variable initialization.
-    */
-    private Runnable mRunnable = new Runnable() {
-        public void run() // should not have to be documented, class is anon.
+     * anon inner in member variable initialization.
+     */
```
Comment (nmancus1) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner.java:26]: Did we need to reformat these for test cases?

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner2.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
+////////////////////////////////////////////////////////////////////////////////
+// Test case file for checkstyle.
+// Created: 2021
+////////////////////////////////////////////////////////////////////////////////
```
Comment (nmancus1) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner2.java:4]: Same as above.

**Review by nmancus1 — CHANGES_REQUESTED** [review]
A few items:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc2.java (hunk 2 above)
Comment (bossever) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc2.java:4]: This is because there are tests which are identical, apart from how the scope property is defined in them. Some use the keyword as is (`public`) whereas some use this method invocation (`Scope.PUBLIC.getName()`). 

>tests which are identical 

These are otherwise duplicate tests, which would have to be removed.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java (hunk 5 above)
Comment (bossever) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java:12]: Yes, I thought of resolving this here itself, as a lot of test inputs (for other Javadoc checks also) have these comments, and they are quite confusing. Let me know if I should change these.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodInterfacePrivateMethod.java (hunk 1 above)
Comment (bossever) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodInterfacePrivateMethod.java:7]: This seems like a good idea, I have made the changes.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc3.java (hunk 3 above)
Comment (bossever) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc3.java]: Done.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java (hunk 4 above)
Comment (bossever) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java]: Done.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly3.java (hunk 6 above)
Comment (bossever) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly3.java]: Done.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner2.java (hunk 9 above)
Comment (bossever) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner2.java]: Done.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java (hunk 7 above)
Comment (bossever) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java]: Looks like I overlooked this, fixed now.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner.java (hunk 8 above)
Comment (bossever) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner.java:22]: No, but we might as well have similarly formatted javadoc in all input files, so I tweaked it. Let me know if this change should be reverted.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java (hunk 5 above)
Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java:8]: I think that any comments other than "ok" and "violation" are extraneous in test inputs usually, unless logic for violations in complicated (counting number of occurrences, etc).  Many of the comments in these javadoc checks are because they were used for mutliple tests with different configs; now that you are changing this to one test input per test, these aren't needed.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodInterfacePrivateMethod.java (hunk 1 above)
Comment (nmancus1) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodInterfacePrivateMethod.java:7]: All except this one :)

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly.java [hunk 10]
```diff
@@ -87,16 +86,16 @@ public void method(StringBuffer aA)
 
 
     /**
-       A param tag should not be required here when relaxed about Javadoc.
-       Writing a little documentation should not be worse than not
-       writing any documentation at all.
+     * A param tag should not be required here when relaxed about Javadoc.
+     * Writing a little documentation should not be worse than not
+     * writing any documentation at all.
```
Comment (nmancus1) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly.java:91]: I would probably revert any formatting changes or put them in a separate commit.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner.java [hunk 11]
```diff
@@ -1,59 +1,57 @@
-////////////////////////////////////////////////////////////////////////////////
-// Test case file for checkstyle.
-// Created: 2001
-////////////////////////////////////////////////////////////////////////////////
```
Comment (nmancus1) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner.java:4]: Removal of these + addition of `/* Config` comment with an extra line might have prevented changing of violation line numbers in UT file :) No need to change now, but maybe remember for the future to save time.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner.java (hunk 8 above)
Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner.java:22]: I would keep all reformatting to separate commit or revert, it just helps reviews to go faster when there is less changes to look at, or when they are obviously atomic.

**Review by nmancus1 — CHANGES_REQUESTED** [review]
Few more:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc2.java (hunk 2 above)
Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc2.java]: If we have more tests that only differ by a string literal used in a property, and a method that returns the same value, please remove them.  The runtime does not know the difference between "public" and `Scope.PUBLIC.getName()` once compiled. CI will show if we needed these tests for some reason.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java (hunk 7 above)
Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java (hunk 4 above)
Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc3.java (hunk 3 above)
Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc3.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner2.java (hunk 9 above)
Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner2.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly3.java (hunk 6 above)
Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly3.java]: Resolved.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodInterfacePrivateMethod.java (hunk 1 above)
Comment (bossever) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodInterfacePrivateMethod.java:7]: Done.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc2.java (hunk 2 above)
Comment (bossever) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc2.java]: Yeah, I was just a little unsure, since the test names differred by name `testX` vs `testXWithResolver`. I have removeed the duplicate now.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java (hunk 5 above)
Comment (bossever) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java]: Removed the extra comments.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner.java (hunk 8 above)
Comment (bossever) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner.java:22]: Moved reformatting to new commit.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly.java (hunk 10 above)
Comment (bossever) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly.java:82]: Moved reformatting to new commit.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodInterfacePrivateMethod.java (hunk 1 above)
Comment (nmancus1) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodInterfacePrivateMethod.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc2.java (hunk 2 above)
Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodNoJavadoc2.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java (hunk 5 above)
Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly2.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner.java (hunk 8 above)
Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner.java:22]: Resolved.

**Review by nmancus1 — APPROVED** [review]
Duplicate comments that appeared above are from replying to old review items WHILE making a new review, sorry for all the noise.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly.java (hunk 10 above)
Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodPublicOnly.java:82]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner.java (hunk 11 above)
Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocmethod/InputMissingJavadocMethodScopeAnonInner.java:4]: Resolved.

**Review by romani — APPROVED** [review]
